src/examples/zkapps/zkapp-self-update.ts

```diff
@@ -59,7 +59,29 @@ Provable.log('original verification key', fooVerificationKey);
 
 const { verificationKey: barVerificationKey } = await Bar.compile();
 
+try {
+  const illegalVerificationKey = new VerificationKey({
+    data: fooVerificationKey!.data,
+    hash: barVerificationKey.hash,
+  });
+
+  const tx2 = await Mina.transaction(deployer, async () => {
+    // VK with mismatched hash and data should throw
+    await contract.replaceVerificationKey(illegalVerificationKey);
+  });
+  await tx2.prove();
+} catch (error: any) {
+  if (
+    error.message.includes('The verification key hash is not consistent with the provided data')
+  ) {
+    console.log('correctly threw on illegal verification key');
+  } else {
+    throw error;
```

Comment:
```suggestion
    throw new Error(`Unexpected error: ${error.message}`);
```
This would only be thrown when an unexpected error occurs. It could be better to be more explicit about this.